### PR TITLE
Adding a thanks page

### PR DESCRIPTION
**Pull request by csomor**
take the situation of the offered macstadium resources to thank our infrastructure sponsors on a separate page

### CODE REVIEW

File: thanks/index.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+</div>
+
+{::options parse_block_html="false" /}
+
+[GitHub]: https://github.com
+[GitHubLogo]: GitHub_Logo.png 
+{: width="200px"}
+
+[MacStadium]: https://www.macstadium.com
+[MacStadiumLogo]: https://uploads-ssl.webflow.com/5ac3c046c82724970fc60918/5c019d917bba312af7553b49_MacStadium-developerlogo.png
```
Comment (vadz) [inline thanks/index.md:37]: Perhaps we should copy their logo here too? This doesn't seem like a permanent/stable URL...

**Review by vadz — APPROVED** [review]
Thanks, no real comments, let's wait for a bit to see if anybody else sees any possible improvements and let's merge it in a couple of days if there are no comments from anybody.

File: thanks/index.md (hunk 1 above)
Comment (csomor) [reply thanks/index.md]: good idea, thanks

**Comment by tierra** [comment]
I wonder if the existing "Credits" page should just be moved to this. I believe this is mostly the same idea behind the credits page, it's just not quite tailored like a "sponsors" page.

https://www.wxwidgets.org/about/credits/

I realize that at least part of the goal here is a more prominent page, linked directly from the top level menu. So rather than suggesting that you use the existing Credits page for these, that's why I'm thinking we actually move (and update) the Credits page to be included in this "Sponsors/Thanks" page.

**Comment by doscott** [comment]
Personally, I think the Credits and Thanks look better the way this PR presents them.

If the credits are updated, in the Hosting section:
 - York University FTP site: Chris Elliott should get credit, but I believe the line should be past tense?
 - Robin Dunn: should get credit, but do we still have a SVN repository and is that where the issue tracker is hosted?
 - Are there still mirror sites?

I ask because I don't see any links to them. Also, I don't see any credits for hosting track, the wiki, on-line doc's.

In Developers their should be an entry for wxWidgets 3

**Comment by csomor** [comment]
> I wonder if the existing "Credits" page should just be moved to this. I believe this is mostly the same idea behind the credits page, it's just not quite tailored like a "sponsors" page.
> 
> https://www.wxwidgets.org/about/credits/
> 
> I realize that at least part of the goal here is a more prominent page, linked directly from the top level menu. So rather than suggesting that you use the existing Credits page for these, that's why I'm thinking we actually move (and update) the Credits page to be included in this "Sponsors/Thanks" page.

@tierra Good idea, I've tried to merge this - left the old credits still there for the moment, so that we can compare ...

**Comment by csomor** [comment]
> Personally, I think the Credits and Thanks look better the way this PR presents them.
> 
> If the credits are updated, in the Hosting section:
> 
> * York University FTP site: Chris Elliott should get credit, but I believe the line should be past tense?
> * Robin Dunn: should get credit, but do we still have a SVN repository and is that where the issue tracker is hosted?
> * Are there still mirror sites?
> 
> I ask because I don't see any links to them. Also, I don't see any credits for hosting track, the wiki, on-line doc's.
> 
> In Developers their should be an entry for wxWidgets 3

very good points Danny, I'm afraid I don't know too much about the current infrastructure situation outside the CI and github ... @vadz @tierra ?

**Comment by vadz** [comment]
FTP site doesn't exist since many years, and while it would be nice to thank all the past sponsors etc, we're going to have to make a much larger list of them to include everybody since 30 years... And while I am/we are definitely thankful to Robin, and especially Bryan, for maintaining wxTrac and wxWiki, it's a bit weird to thank ourselves, basically.

**Comment by tierra** [comment]
Indeed. I've been hosting Trac and (read-only) SVN for a while now, in addition to the wiki, forums, docs, and this site that I've been hosting for much longer. But all of it is (still) funded through old GSoC stipends. I don't mind whether I'm mentioned or not in credits page, some of it is already mentioned on the team page anyway.

**Comment by csomor** [comment]
Ok, I've taken out the old parts, I agree that we must clean from time to time, otherwise the page will not be read ...

@vadz I agree that we should add a line or two for wxWidgets 3, otherwise that part looks very much outdated

@tierra, @vadz the rust project has quite an interesting automated Thank you page for contributing, https://thanks.rust-lang.org does anybody have an idea whether we could do something similar ?

**Comment by vadz** [comment]
> @tierra, @vadz the rust project has quite an interesting automated Thank you page for contributing, https://thanks.rust-lang.org does anybody have an idea whether we could do something similar ?

This is probably just based on `git shortlog -s` output. At least this is what I use for the commits/contributors statistics included in the release announcements.

File: thanks/index.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+[AppVeyor]: https://www.appveyor.com
+
+### Software and Financial Donations
+
+[VMware, Inc.][VMware] has supplied free VMware licences to members of the wxWidgets
+development team. VMware is a superb solution for people who need to access
+more than one operating system simultaneously. A big thank you to VMware for
+supporting the project.
+
+[VMware]: http://www.vmware.com/
```
Comment (PBfordev) [inline thanks/index.md:56]:
```suggestion
[VMware]: https://www.vmware.com/
```

File: thanks/index.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+Thank you to [Microsoft][] for donating a copy of Visual C++ 6.0 to help
+wxWidgets compile on this version of the compiler (for a Virginia Tech course).
+
+[Microsoft]: http://www.microsoft.com/
```
Comment (PBfordev) [inline thanks/index.md:61]:
```suggestion
[Microsoft]: https://www.microsoft.com/
```

File: thanks/index.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+[H2Reg]: http://www.helpware.net/mshelp2/h2reg.htm
+
+#### Individual Donations
+
+* Andy Robinson, [Seventh String Software][]
+* Alec Ross, Arcs Ltd.
+* Sandro Sigala, Roxantis
+
+[Seventh String Software]: http://www.seventhstring.com/
```
Comment (PBfordev) [inline thanks/index.md:83]:
```suggestion
[Seventh String Software]: https://www.seventhstring.com/
```

File: thanks/index.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+### Hosting
+
+The stipends from our participation in [Google Summer of Code][GSoc] allow Bryan Petty to finance the hosting for this website, the online user manuals, wxTrac, the read-only SVN and the wxWiki.
```
Comment (PBfordev) [inline thanks/index.md:87]:
```suggestion
The stipends from our participation in [Google Summer of Code][GSoC] allow Bryan Petty to finance the hosting for this website, the online user manuals, wxTrac, the read-only SVN and the wxWiki. 
```

File: thanks/index.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+Systä, Arthur Tetzlaff-Deas, Jonathan Tonberg, Jyrki Tuomi, Janos Vegh, Andrea
+Venturoli, David Webster, Xiaokun Zhu, Edward Zimmermann.
+
+### Other Support
+
+We would like to thank [AIAI][7], University of Edinburgh (and in particular
+Austin Tate and Robert Rae) for supporting the original work and giving
+permission to liberate the wxWidgets source.
+
+[7]: http://www.aiai.ed.ac.uk/
```
Comment (PBfordev) [inline thanks/index.md:144]: Perhaps this URL would be better
https://web.inf.ed.ac.uk/aiai

**Comment by csomor** [comment]
> I have mostly just changed the protocol from http to https where possible (i.e., I could access the web).

thanks a lot

File: thanks/index.md (hunk 6 above)
Comment (csomor) [reply thanks/index.md:144]: yes, thanks, I'll use that

**Comment by vadz** [comment]
Assuming we've accurately recorded the authorship of all commits since the switch to Git (which is, unfortunately, not an accurate assumption, I'm afraid), the full list of contributors since the switch from svn can be found by running
```
$ git shortlog -s 0df45db7f66b5aa0ddc6043b794cc19f5522c033..|cut -f2
```

Such list is, however, going to fatally going to get out of date very soon, so maybe we should use some cut-off, e.g. 10 commits at a minimum? This would bring down the number of people produced by the command above from 351 (with some duplicates...) to 56, which seems much more reasonable, but maybe too stingy? Using 5 as a cut off would give 82 contributors, and 3 132.

Alternatively, we could just link to https://github.com/wxWidgets/wxWidgets/graphs/contributors, but it just seems right to have a list (or at least a link to the list) of more recent contributors too, if we have them for wx 2.

File: thanks/index.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+* wxWidgets 2 for Motif: Julian Smart - didn't quite mean to do that port too,
+  oh well
+* wxWidgets 2 for Mac: Stefan Csomor - keeping baby/keyboard separation optimal
+* wxWidgets 1 for Xt: Markus Holzem - keeping his sanity now he's away
+* wxString, wxConfig, Windows drag and drop, and loads of other great stuff
+  without which we would be quite stuck: the one and only Vadim Zeitlin,
+  physicist and wxWidgets hero
+* wxThread, wxSocket and multimedia classes: the deceptively young but talented
+  Guilhem Lavaux
+* wxHTML: the highly dedicated Vaclav Slavik
```
Comment (PBfordev) [inline thanks/index.md:127]: Another nitpick: Václav's name is listed in wxWidgets 3 contributors with diacritics, and he uses diacritics not only on GitHub but also on his English-written web homepage. Perhaps the same form of his name should be used everywhere in this document.
```suggestion
* wxHTML: the highly dedicated Václav Slavík
```

**Comment by csomor** [comment]
> Assuming we've accurately recorded the authorship of all commits since the switch to Git (which is, unfortunately, not an accurate assumption, I'm afraid), the full list of contributors since the switch from svn can be found by running
> 
> ```
> $ git shortlog -s 0df45db7f66b5aa0ddc6043b794cc19f5522c033..|cut -f2
> ```
> 
> Such list is, however, going to fatally going to get out of date very soon, so maybe we should use some cut-off, e.g. 10 commits at a minimum? This would bring down the number of people produced by the command above from 351 (with some duplicates...) to 56, which seems much more reasonable, but maybe too stingy? Using 5 as a cut off would give 82 contributors, and 3 132.
> 
> Alternatively, we could just link to https://github.com/wxWidgets/wxWidgets/graphs/contributors, but it just seems right to have a list (or at least a link to the list) of more recent contributors too, if we have them for wx 2.

Thanks a lot, I've added a wx3 list capping at 5 and a link to contributors

File: thanks/index.md (hunk 7 above)
Comment (csomor) [reply thanks/index.md]: Thanks

**Comment by vadz** [comment]
I didn't realize this hasn't been merged yet, so I've just (squash) merged it, thanks!

**Pull request closed without merging**